### ISSUE TITLE
Clarifying steps to move things along

### ISSUE THREAD

**Issue by knghtbrd** [xapp-project/xapp-project#3]
Posted as an issue to avoid adding to the noise of the PR.

At the time of writing, this repository has a kind of a sparse readme, a single PR #1 with a thin spec that's kind of a good start draft, and some discussion about it which seems to be slowing down with almost consensus on what's there. While I'm _barely_ qualified to call myself any kind of GUI developer (and thus wholly unqualified to make decisions or even **help** organize any of this), I'd like help get things moving again.

What do we need to approve the current PR, be it as a first version of an XApp standard, or a draft for same? It might seem like it's got a consensus for what's there to me, but … who's making that call? Either way, I think it'll be much easier for people to make individual proposals once there's some kind of status quo to propose against, and proposals might more easily have their own discussions that way.

What it takes to approve or reject a proposal and who gets to decide is the other question that should be taken up pretty soon. Debian does its policy by consensus unless there's some hard question or split that warrants a vote. But in the Linux world there are some folks determined to do things their own way and who won't have it any other.  One might argue this whole project exists because Gnome is built around the concept of dying on that hill, but they're hardly alone. Some thought to that should be given, if it hasn't been yet.

These aren't the only questions to answer, but hopefully they'll be enough to get things moving again. 🙂

**Comment by alexxcons** [comment]
> What do we need to approve the current PR, be it as a first version of an XApp standard, or a draft for same?

Several things will need some kind of consensus before merge:

- [This thread](https://github.com/xapp-project/xapp-project/pull/1#discussion_r1631545766) about window decorations
  - IMO we should either have "CSD support is optional", or we should have "An XApp has to support CSD and SSD".
  -  If we allow CSD-only Apps, we need to modify the [README snippet ](https://github.com/xapp-project/xapp-project/pull/1/files#diff-e7e2fa8723d3a4c69e37b7668c9d2e7550513164fe1a17efaaaf4efce0d8f2e9R64)a bit, since good integration gets problematic for some DE's.
  - ~~It would be nice to have a global XSetting "preferCSD". So that there is no need to define the default per App. (Though not sure about an equivalent of XSettings in wayland)~~ Edit: As Brian stats in the thread, just obey to what the used window manager requests.
  - @clefebvre, IMO it would be nice if you could just make some decision on this one, in order to move forward.
- [This thread]( https://github.com/xapp-project/xapp-project/pull/1#discussion_r1631997225) about theming
  - Leave it out for the first, or write  "If the used toolkit offers the possibility, xapps should be themable" ?
  - @clefebvre, would be nice if you could make a decision on this one as well (I am not that much into theming anyway)

Besides that, I think the main targets are clear enough / it would be fine to be merged. Some rather cosmetic things could be discussed/added later, e.g. what was suggested by @clefebvre in [this comment](https://github.com/xapp-project/xapp-project/pull/1#pullrequestreview-2118227526)

> Either way, I think it'll be much easier for people to make individual proposals once there's some kind of status quo to propose against, and proposals might more easily have their own discussions that way.

I agree on that ... once we have a spec, we most likely will see people which want to add software/mirrors to The XApp projects. (Actually I am waiting on it myself, to possibly add some Xfce mirrors)

> What it takes to approve or reject a proposal and who gets to decide is the other question that should be taken up pretty soon. 

Yea, I suppose we should as well add a paragraph about that, though I don't see an urgency for it.
  - Who has to agree on changes in the spec
  - Who has to agree on new projects which are to be added / existing projects which should be removed

For now, I suppose it will be fine to have @clefebvre as benevolent dictator.

**Comment by knghtbrd** [comment]
> Several things will need some kind of consensus before merge:
> […]

The major points have probably all been well-made on the SSD/CSD issue. I think we're mostly in agreement that SSD needs to be supported, it's just a question of if or how strong we want to support CSD. Otherwise I agree, we just need Clem to make a call on that one. If things change later and it's the wrong call down the line, they do. I don't think anyone advocates for a MUST NOT on the question of supporting CSD unless it be on their own desktop. 😉

I brought up my own "cosmetic" issue a few weeks ago. There is no love lost between me and the Gnome core devs. I'll spare you the rant about discrimination and mockery of my disability. It was a decade ago to the month actually and I still feel nothing but rage about it. Be that as it may, the spec should not be "Gnome bad!" But I don't see that as cosmetic so much as it is that this spec, and what I hope will become a standards group in time, should be about defining what we want for the bendefit of all Linux users, not defining what pisses us off about a single project that does not play well with others.

> > What it takes to approve or reject a proposal and who gets to decide is the other question that should be taken up pretty soon.
> 
> Yea, I suppose we should as well add a paragraph about that, though I don't see an urgency for it.
> 
>     * Who has to agree on changes in the spec
>     * Who has to agree on new projects which are to be added / existing projects which should be removed
> 
> For now, I suppose it will be fine to have @clefebvre as benevolent dictator.

God help us.

Er, I mean, oh, okay! 😁 Fine with Clem having control of things at this stage. But again, I hope this effort will grow to be significant enough that honestly his time would be better spent on other projects. And while I really don't want to distract from the effort to get what's already there merged into the repo, there'll come a point where those questions of who's vote does or doesn't count will matter. In the same vein, it's inevitable that there'll come a point where someone will be more focused on derailing the effort than supporting it. Because The Internet, naturally. It's not a priority now, thankfully.
